src/odemis/driver/smaract.py

```diff
@@ -3084,3 +3085,1139 @@ def SA_CTL_Move(self, handle, ch, pos_pm, _):
+                                     SA_SIDLL.SA_PS_ADJUSTMENT_STATE_MANUAL_ADJUST)
+                self.WaitForEvent(SA_SIDLL.SA_PS_ADJUSTMENT_STATE_MANUAL_ADJUST)
+
+                if future._must_stop:
+                    raise CancelledError()
+                # Activate working distance
+                self.SetProperty_i32(SA_SIDLL.SA_PS_SYS_WORKING_DISTANCE_ACTIVATE_PROP,
+                                     SA_SIDLL.SA_PS_WORKING_DISTANCE_SHRINK_MODE_LEFT_RIGHT)
+                # attribute is write-only and there is no corresponding event type (yet some waiting time is necessary)
+                # --> wait 1 s for command to be processed
+                time.sleep(1)
+
+                if future._must_stop:
+                    raise CancelledError()
+                # Switch to automatic adjustment
+                self.SetProperty_i32(SA_SIDLL.SA_PS_AF_ADJUSTMENT_STATE_PROP,
+                                     SA_SIDLL.SA_PS_ADJUSTMENT_STATE_AUTO_ADJUST)
+                self.WaitForEvent(SA_SIDLL.SA_PS_ADJUSTMENT_STATE_DISABLED)  # state will be DISABLED when done
+                logging.debug("Finished referencing.")
+
+                # We could save the referencing parameters to memory here. However, it is only possible
+                # to save these parameters together with the configuration parameters (fiber length etc.).
+                # In case something goes wrong, there is a chance that the device needs to be reconfigured.
+                # Since referencing should be done on startup anyway, there is no real need to save the
+                # parameters.
+            except SA_SIError as ex:
+                # This occurs if a stop command interrupts referencing
+                if ex.errno == SA_SIDLL.SA_SI_ERROR_CANCELLED:
+                    future._was_stopped = True
+                    logging.info("Referencing stopped: %s", ex)
+                    raise CancelledError()
+                elif future._must_stop:
+                    future._was_stopped = True
+                    raise CancelledError()
+                else:
+                    logging.error("Referencing failed: %s", ex)
+                    raise
+            except CancelledError:
+                logging.debug("Referencing cancelled.")
+                future._was_stopped = True
+                raise  # No fuss, pass it as-is
+            except Exception:
+                logging.exception("Referencing failure.")
+                raise
+            finally:
+                # If the state of the device is stable and all channels are valid, the referencing procedure
+                # succeeded. Typically, all channels become valid at the same time during the reference procedure.
+                # We still check all of them inidividually to be sure.
+                if self.IsStable():
+                    for ch, num in self.channels.items():
+                        self.referenced._value[ch] = self.IsValid(num)
```

Comment:
Could we set the initial referencing to this? If not, it probably shouldn't be in the "finally".

Reply:
What I mean, is that for now, `.referenced` is *always* set to `False` at init. If the hardware has already been referenced + validated, and then backend restarted, it'd be nice to immediately set .referenced to `True`. Can we do that by using `IsValid()`?